tests/test_service/test_adapter.py

```diff
@@ -0,0 +1,142 @@
+"""End-to-end tests of the SMAP-L2-Gridding-Service."""
+
+import pytest
+from harmony_service_lib.message import Message as HarmonyMessage
+from harmony_service_lib.util import config
+from xarray import open_datatree
+
+from harmony_service.adapter import SMAPL2GridderAdapter
+from smap_l2_gridder.exceptions import InvalidGPDError
+
+
+def test_process_sample_file(tmp_path, sample_datatree_file, stack_catalog, mocker):
+    """Run a sample file through the adapter."""
+    # override the adapter's working dir
+    temp_dir_mock = mocker.patch('harmony_service.adapter.TemporaryDirectory')
+    temp_dir_mock.return_value.__enter__.return_value = tmp_path
+
+    # use a datatree fixture as the downloaded file
+    download_mock = mocker.patch('harmony_service.adapter.download')
+    download_mock.return_value = sample_datatree_file
+
+    # set the output filename
+    filename_mock = mocker.patch('harmony_service.adapter.generate_output_filename')
+    output_filename = tmp_path / 'test-gridded.nc'
+    filename_mock.return_value = output_filename
+
+    stage_mock = mocker.patch('harmony_service.adapter.stage')
+    staging_dir = tmp_path / 'staging'
+
+    message = HarmonyMessage(
+        {
+            'accessToken': 'fake_token',
+            'callback': 'https://example.com/',
+            'sources': [{'collection': 'C1234-EEDTEST', 'shortName': 'smap_test'}],
+            'stagingLocation': staging_dir,
+            'user': 'fakeUsername',
+            'format': {'mime': 'application/x-netcdf4'},
+        }
+    )
+
+    # Set up Adapter class
+    smap_l2_gridding_service = SMAPL2GridderAdapter(
+        message, config=config(validate=False), catalog=stack_catalog
+    )
+
+    # Invoke the adapter.
+    _, _ = smap_l2_gridding_service.invoke()
```

Comment:
I kind of thought they were useless but they can be added.  

I checked, and there's nothing of interest in the output stac catalog because everything is mocked beyond.  The interesting stuff is that we actually call the gridding logic and can open the file.  That said, I mocked the Asset call to add one more check.